tests/test_admin.py

```diff
@@ -229,9 +231,25 @@ class AdminRegisterVersionTestCase(CMSTestCase):
     def test_register_version_admin(self):
         """Test that VersionAdmin is registered for Version model
         """
+        site = admin.AdminSite()
 
-        self.assertIn(Version, admin.site._registry)
-        self.assertEqual(admin.site._registry[Version].__class__, VersionAdmin)
+        register_versionadmin_proxy(PollContent, 'foo', site)
+
+        self.assertIn(PollContent, site._registry)
+        self.assertIn(VersionAdmin, site._registry[PollContent].__class__.mro())
+
+    def test_register_version_admin_again(self):
+        """Test that attempting to register a proxy model again
+        doesn't do anything.
+        """
+        existing_admin = type('TestAdmin', (admin.ModelAdmin, ), {})
+        site = admin.AdminSite()
+        site.register(PollContent, existing_admin)
+
+        with patch.object(site, 'register') as mock:
+            register_versionadmin_proxy(PollContent, 'foo', site)
+
+        mock.assert_not_called()
```

Comment:
If a warning is added then this should probably also test that the warning is being logged.